File: 프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people, limit):
    people.sort()
    left = 0
    right = len(people) - 1
    answer = 0
    while left <= right :
        boat = limit
        if people[right] + people[left] <= limit :
            right -= 1
            left += 1
        else :
            right -= 1
        answer += 1
    
    return answer

File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.io.*;
import java.util.*;
class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        boolean[] visited = new boolean[people.length];
        
        Arrays.sort(people);
        
        int left = 0;
        int right = people.length - 1;
        
        while(left <= right){
            answer++;
            if(people[left] + people[right] <= limit){
                left ++;
                right --;
            }
            else
                right--;
        }
        return answer;
    }
    
}